otelcol/command.go

```diff
@@ -52,10 +67,11 @@ func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet) error
 	if len(resolverSet.URIs) == 0 {
 		return errors.New("at least one config flag must be provided")
 	}
-	// Provide a default set of providers and converters if none have been specified.
-	// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
-	// do it in the builder.
+
 	if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
+		if enforceProviders {
+			return errors.New("at least one provider or converter must be provided")
```

Comment:
```suggestion
			return errors.New("at least one Provider must be supplied")
```
I know this wording is weird since we're checking for both providers and converters, but we should avoid implying that only supplying a Converter is a valid state. If a user providers only a Converter, they'll get a different error (from the Resolver), but it will at least say the same thing.